packages/remix-dev/compiler.ts

```diff
@@ -299,7 +310,7 @@ async function buildEverything(
     // it when loading the @remix-run/dev/assets-manifest virtual module.
     assetsManifestPromiseRef.current = assetsManifestPromise;
 
-    let serverBuildPromise = createServerBuild(
+    let serverBuildPromise = await createServerBuild(
```

Comment:
We shouldn't await `createServerBuild` here, as it's already in the `Promise.all` we're `await`ing.

```suggestion
    let serverBuildPromise = createServerBuild(
```

Reply:
Fixed :+1: